Feature/blt phpcbf

dooleymatt opened the pull request:
Fixes # 977.

Changes proposed:
- Adds `fix:phpcbf` target to automatically fix violations found by phpcs.

dooleymatt commented:
@grasmash this is ready to be reviewed. What action should I take to indicate that?

grasmash commented:
@dooleymatt Please add a test for this feature. Ideally, you should have a tests that asserts both positive and negative cases.

dooleymatt commented:
Thanks @grasmash. I took a look at the existing tests to see how to write one for this, but I'm not really sure what I should I be testing.  It seems like most of the existing tests check for files that have been added, removed, or changed, but I'm not sure what similar tests could be performed for this new task. I am hoping either you or @TravisCarden could offer some direction. Thanks!

dooleymatt commented:
@grasmash Test has been added.

dooleymatt commented:
@TravisCarden Thanks for reviewing this. I initially didn't include a test like you described because I felt like it might have been outside the scope of this to test whether or not phpcbf actually does what it's supposed to do. However, I've updated the test to include that now, and have also updated the way that the phpcs csv file report is generated.

grasmash commented:
@dooleymatt This looks great. My only request is that we put this in the `validate:` namespace rather than creating a new `fix:` namespace.

dooleymatt commented:
@grasmash I based that change on Travis's request here: https://github.com/acquia/blt/issues/977#issuecomment-274567320

grasmash commented:
Eh, alright. The argument makes sense.
